src/App.jsx

```diff
@@ -1,12 +1,65 @@
 import './App.css'
+import logo from './assets/pokemon_logo.jpg'
+import {useEffect, useState} from "react";
+import Button from "./components/button/Button.jsx";
+import Pokemon from "./components/pokemon/Pokemon.jsx";
+import axios from "axios";
 
 function App() {
+    const [pokemon, setPokemon] = useState([])
+    const [error, toggleError] = useState(false)
+    const [loading, toggleLoading] = useState(false)
+    const [endpoint, setEndpoint] = useState('https://pokeapi.co/api/v2/pokemon/')
 
-  return (
-    <>
-      <h1>Gotta catch em all!</h1>
-    </>
-  )
+    useEffect(() => {
+        const controller = new AbortController()
+
+        async function fetchData() {
+            toggleLoading(true)
+            toggleError(false)
+
+            try {
+                const {data} = await axios.get(endpoint, {
+                    signal: controller.signal
+                })
+                setPokemon(data);
+            }
+            catch (e){
+                if (axios.isCancel(e)) {
+                    console.error('Request is canceled...')
+                }
+                else {
+                    console.error(e)
+                    toggleError(true)
+                }
+            }
+            finally {
+                toggleLoading(false)
+            }
+        }
+
```

Comment:
Het request wordt nooit gecancelled. Als je aan het eind van de useEffect function body een functie teruggeeft, dan wordt die altijd aangeroepen vóórdat de useEffect weer aangeroepen wordt:
```suggestion
        return () => {
            controller.abort();
        }
```
Zo wordt altijd eerst het lopende request gecancelled voordat een nieuw request afgevuurd wordt.